DISCOVER/404.html

```diff
@@ -0,0 +1,163 @@
+  <div class="glass-card">
+    <h1 id="title">404 - Page Not Found</h1>
+    <p id="message">This page might’ve wandered off or was never here at all.</p>
+    <p id="suggestion">Maybe it was deleted, renamed, or never existed in the first place.</p>
+    <p><strong id="what-do">What can you do?</strong></p>
+    <a href="./intro.html" id="home">⬅️ Back to Home</a>
+    <a href="https://github.com/numfocus/DISCOVER-Cookbook/issues" id="report">🐞 Report the Issue</a>
+
+    <div>
+      <label for="language-switcher" style="display:block; margin-top:1.5rem; color:#444;">🌐 Switch Language:</label>
+      <select id="language-switcher">
+        <option value="en">English</option>
+        <option value="es">Español</option>
+        
+      </select>
+    </div>
+
+    <p id="footer" style="margin-top: 1.2rem;">If you think this is an error, let us know by creating an issue.</p>
+  </div>
+
+  <script>
+    const translations = {
+      en: {
+        title: "404 - Page Not Found",
+        message: "This page might’ve wandered off or was never here at all.",
+        suggestion: "Maybe it was deleted, renamed, or never existed in the first place.",
+        whatDo: "What can you do?",
+        home: "⬅️ Back to Home",
+        report: "🐞 Report the Issue",
+        footer: "If you think this is an error, let us know by creating an issue."
+      },
+      es: {
+        title: "404 - Página no encontrada",
+        message: "Esta página se ha perdido o nunca existió.",
+        suggestion: "Tal vez fue eliminada, renombrada o nunca existió.",
+        whatDo: "¿Qué puedes hacer?",
+        home: "⬅️ Volver al inicio",
+        report: "🐞 Informar del problema",
+        footer: "Si crees que esto es un error, infórmanos creando un issue."
+      }
+      
+    };
+
+    const pathLang = location.pathname.split("/")[1];
+    const refLang = new URL(document.referrer || "", location.origin).pathname.split("/")[1];
+    const lang = translations[pathLang] ? pathLang : (translations[refLang] ? refLang : "en");
+
+
+    document.getElementById("title").textContent = translations[lang].title;
+    document.getElementById("message").textContent = translations[lang].message;
+    document.getElementById("suggestion").textContent = translations[lang].suggestion;
+    document.getElementById("what-do").textContent = translations[lang].whatDo;
+    document.getElementById("home").textContent = translations[lang].home;
+    document.getElementById("report").textContent = translations[lang].report;
+    document.getElementById("footer").textContent = translations[lang].footer;
+    document.getElementById("language-switcher").value = lang;
+
+    document.getElementById("language-switcher").addEventListener("change", function () {
+        const selectedLang = this.value;
+        if (selectedLang === "en") {
+            window.location.href = `/404.html`;
+        } else {
+          window.location.href = `/${selectedLang}/404.html`;
+    }
+  });
```

Comment:
or maybe this isn't updating the language related constants? not even sure it is possible though (0 javascript knowledge on my end)